data-processing-lib/ray/src/data_processing_ray/runtime/ray/ray_utils.py

```diff
@@ -113,7 +113,7 @@ def operator() -> ActorHandle:
         actors = [operator() for _ in range(n_actors)]
         for i in range(120):
             time.sleep(1)
-            alive = list_actors(filters=[("class_name", "=", cls_name), ("state", "=", "ALIVE")])
+            alive = list_actors(filters=[("class_name", "=", cls_name), ("state", "=", "ALIVE")], limit=n_actors + 10)
```

Comment:
Why is 10 the right number. what if a transform's ray runtime starts its own actors?  Maybe this should be n_actors*2 or the above with the ability to override with a config value (env var?)

Reply:
I agree.  The idea to use the number of actors to compute the limit is great, but l think it would be better to use 
```
num_actors * 10
```
instead of 
```
num_actors + 10
```

Reply:
what is the downside of setting the limit to 1E6

Reply:
Gents, If we create 5 actors whats the reason asking for 100000? We are asking only for what we need and padding it with 10

Reply:
@blublinsky 1) what is the downside of using 1E6 and 2) what if the transform creates its own actors?  Please advise.

Reply:
take a closer look at this line of code.
We are listing only actors, based on filters - class name and running. I do not see transform creating actors using the same class. I do not want to put a number of 1e6, which is an absolutely arbitrary number. This said, I assume that if an transform creates his own actors, it will use the same API, but listing in this case will use a different filter.

Reply:
@blublinsky : I have two questions about the code we are discussing here:
https://github.com/IBM/data-prep-kit/blob/91f22eb7f321442e882899804b9cc043ff8eba54/data-processing-lib/ray/src/data_processing_ray/runtime/ray/ray_utils.py#L114-L121
1. It looks like the only way for this code to execute successfully is if the `list_actors()` method returns **exactly** `n_actors` not more and not less.  Then, I think the limit should equal `n_actors`
2. I might be missing something, but why is the condition on line 117 requiring that all the actors are up?  Wouldn't this work even if, say, 80% of the actors were up?  In other words, what the drawback on replacing the condition on line 117 with this one?
```
if len(alive) >= 0.8 * len(actors):
    return actors
# failed - ....
```

Reply:
I agree, but I don't understand why we need to add 10 to the number of actors in the limit. Why not just set the limit to be equal to the number of actors, since this is what is tested on line 117. 

For example, if there are 5 actors, why would we ever need a limit greater than 5? Can it happen that `list_actors()` will return 6 actors, if we only intended to create 5?  And, in that case, do we need the run to fail because there are more actors available than what we requested to create?

Reply:
Just in case, I am  a bit concerned to use the same number

Reply:
Another thing that I found today, max limit supported is 10000

Reply:
@cmadam, @daw3rd any progress on this?

Reply:
If the number of created actors is greater than the limit, the function `create_actors()` should still return the actor handles